src/main/java/com/amplitude/api/DeviceInfo.java

```diff
@@ -200,6 +202,30 @@ private String getAdvertisingId() {
             }
         }
 
+        private String getAppSetId() {
+            try {
+                Class AppSet = Class
+                        .forName("com.google.android.gms.appset.AppSet");
+                Method getClient = AppSet.getMethod("getClient", Context.class);
+                Object appSetClient = getClient.invoke(null, context);
+                Method getAppSetInfo = appSetClient.getClass().getMethod("getAppSetInfo");
+                Object taskWithAppSetInfo = getAppSetInfo.invoke(appSetClient);
+                Class Tasks = Class.forName("com.google.android.gms.tasks.Tasks");
+                Method await = Tasks.getMethod("await", Class.forName("com.google.android.gms.tasks.Task"));
+                Object appSetInfo = await.invoke(null, taskWithAppSetInfo);
+                Method getId = appSetInfo.getClass().getMethod("getId");
+                appSetId = (String) getId.invoke(appSetInfo);
+            } catch (ClassNotFoundException e) {
+                AmplitudeLog.getLogger().w(TAG, "Google Play Services SDK not found!");
+            } catch (InvocationTargetException e) {
+                AmplitudeLog.getLogger().w(TAG, "Google Play Services not available");
+            } catch (Exception e) {
+                AmplitudeLog.getLogger().e(TAG, "Encountered an error connecting to Google Play Services for app set id", e);
```

Comment:
The message in the Advertising ID error should be changed to reflect Ad ID and the new situation around that. Ad ID may not be available for a good reason (user opts out), however, our SDK won't know the exact reason

Reply:
Oh, I can add the ad id in the ad id function as well, so we tell the difference from the message

Reply:
just update the messages for ad id and app set id for these error messages, it may be confusing to see these messages when check the log.